src/sentry/integrations/utils/code_mapping.py

```diff
@@ -516,34 +524,12 @@ def get_sorted_code_mapping_configs(project: Project) -> list[RepositoryProjectP
     return sorted_configs
 
 
-def _get_code_mapping_source_path(src_file: str, frame_filename: FrameFilename) -> str:
-    """Generate the source code root for a code mapping. It always includes a last backslash"""
-    source_code_root = None
-    if frame_filename.frame_type() == "packaged":
-        if frame_filename.dir_path != "":
-            # src/sentry/identity/oauth2.py (sentry/identity/oauth2.py) -> src/sentry/
-            source_path = src_file.rsplit(frame_filename.dir_path)[0].rstrip("/")
-            source_code_root = f"{source_path}/"
-        elif frame_filename.root != "":
-            # src/sentry/wsgi.py (sentry/wsgi.py) -> src/sentry/
-            source_code_root = src_file.rsplit(frame_filename.file_name)[0]
-        else:
-            # ssl.py -> raise NotImplementedError
-            raise NotImplementedError("We do not support top level files.")
-    else:
-        # static/app/foo.tsx (./app/foo.tsx) -> static/app/
-        # static/app/foo.tsx (app/foo.tsx) -> static/app/
-        source_code_root = f"{src_file.replace(frame_filename.file_and_dir_path, remove_straight_path_prefix(frame_filename.root))}/"
-    if source_code_root:
-        assert source_code_root.endswith("/")
-    return source_code_root
```

Comment:
This function has been replaced by the `find_roots` function, which is now the universal logic to derive stack trace and source code roots for both automatically derived code mappings and those created manually from the `project_repo_path_parsing` endpoint. I 

`find_roots` is a better way to derive the stack trace root and source code root because it is more robust and requires less branching logic to accomodate a specific platform's shenanigans. `find_roots` inherently accommodates both Node and Go's platform-specific stack traces without needing any branching logic or normalization (hence why we can remove `_normalized_stack_and_source_roots`). Hopefully this reduces the complexity and footprint of code-mappings, while being more resilient to other language's shenanigans.